client/scripts/views/pages/profile/validator_profile_stats.ts

```diff
@@ -0,0 +1,239 @@
+import m from 'mithril';
+import { makeDynamicComponent } from 'models/mithril';
+import 'pages/validatorprofile.scss';
+import app from 'state';
+import { ChainBase } from 'models';
+import Substrate from 'controllers/chain/substrate/main';
+import { Coin, formatCoin } from 'adapters/currency';
+import { u32 } from '@polkadot/types';
+import { Card, Spinner, Button } from 'construct-ui';
+import EditProfileModal from 'views/modals/edit_profile_modal';
+import EditIdentityModal from 'views/modals/edit_identity_modal';
+import SubstrateIdentity from 'controllers/chain/substrate/identity';
+import { formatNumberShort } from 'shared/adapters/currency';
+import { ICommissionInfo } from 'controllers/chain/substrate/staking';
+
+const editIdentityAction = (account, currentIdentity: SubstrateIdentity) => {
+  const chainObj = app.config.chains.getById(account.chain);
+  if (!chainObj) return;
+
+  // TODO: look up the chainObj's chain base
+  return (account.chain.indexOf('edgeware') !== -1 || account.chain.indexOf('kusama') !== -1) && m(Button, {
+    intent: 'primary',
+    // wait for info to load before making it clickable
+    class: currentIdentity ? '' : 'disabled',
+    onclick: async () => {
+      app.modals.create({
+        modal: EditIdentityModal,
+        data: { account, currentIdentity },
+      });
+    },
+    label: currentIdentity?.exists ? `Edit ${chainObj.name} identity` : `Set ${chainObj.name} identity`
+  });
+};
+
+export interface IValidatorPageState {
+  dynamic: {
+    validators: any,
+    lastHeader: any,
+    annualPercentRate: ICommissionInfo;
+  };
+  results: any[];
+  identity: SubstrateIdentity | null;
+  fullYearSlashes: any;
+  slashes: any
+  imOnlinePerCent: any,
+  fullYearImOnline: any
+}
+export interface IValidatorAttrs {
+  address: string;
+  total?: Coin;
+  otherTotal?: Coin;
+  bonded?: Coin;
+  commission?: number;
+  toBeElected?: boolean;
+  isOnline?: boolean;
+  hasMessage?: boolean;
+  blockCount?: u32;
+  account: any,
+}
+// getting dates for 1 previous year
+let startDate = new Date();
+startDate.setFullYear(startDate.getFullYear() - 1);
+startDate = new Date(startDate);
+const endDate = new Date();
+export const ValidatorStats = makeDynamicComponent<IValidatorAttrs, IValidatorPageState>({
+  getObservables: (attrs) => ({
+    groupKey: app.chain.class.toString(),
+    validators: (app.chain.base === ChainBase.Substrate) ? (app.chain as Substrate).staking.validators : null,
+    lastHeader: (app.chain.base === ChainBase.Substrate)
+      ? (app.chain as Substrate).staking.lastHeader
+      : null,
+    annualPercentRate: (app.chain.base === ChainBase.Substrate)
+      ? (app.chain as Substrate).staking.annualPercentRate
+      : null
+  }),
+  oncreate: async (vnode) => {
+    m.request({
+      method: 'GET',
+      url: '/api/getSlashes',
```

Comment:
Please follow the structure of the rest of the repo, an example is below even though there are plenty of examples of how to construct the proper route.
```
`${app.serverUrl()}/createComment`
```

Reply:
Thanks for pointing the inconsistency out @drewstone !